src/riak_cs_gc_manager.erl

```diff
@@ -0,0 +1,285 @@
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc All gen_fsm:send_event/2 call should be ignored
+idle(_Event, State) ->
+    {next_state, idle, State}.
+running(_Event, State) ->
+    {next_state, running, State}.
+
+%% @private
+%% @doc
+idle({start, Options}, _From, State) ->
+    MaxWorkers =  riak_cs_gc:gc_max_workers(),
+    %% StartKey = proplists:get_value(start, Options,
+    %%                                riak_cs_gc:epoch_start()),
+    BatchStart = riak_cs_gc:timestamp(),
+    %% EndKey = proplists:get_value('end', Options, BatchStart),
+    Leeway = proplists:get_value(leeway, Options,
+                                 riak_cs_gc:leeway_seconds()),
+
+    %% set many items to GCDState here
+    GCDState = #gc_batch_state{
+                  batch_start=BatchStart,
+                  leeway=Leeway,
+                  max_workers=MaxWorkers},
+
+    case riak_cs_gc_batch:start_link(GCDState) of
+        {ok, Pid} ->
+            {reply, ok, running,
+             State#gc_manager_state{gc_batch_pid=Pid,
+                                    current_batch=GCDState}};
+        Error ->
+            {reply, Error, idle, State}
+    end;
+idle(_, _From, State) ->
+    {reply, {error, idle}, idle, State}.
+
+running(cancel, _From, State = #gc_manager_state{gc_batch_pid=Pid}) ->
+    %% stop gc_batch here
+    catch riak_cs_gc_batch:stop(Pid),
+    NextState=schedule_next(State),
+    {reply, ok, idle, NextState#gc_manager_state{gc_batch_pid=undefined}};
+running({finished, Report}, _From, State = #gc_manager_state{batch_history=H}) ->
+    %% Add report to history
+    NextState=schedule_next(State),
+    {reply, ok, idle,
+     NextState#gc_manager_state{gc_batch_pid=undefined,
+                                batch_history=[Report|H]}};
+running(_Event, _From, State) ->
+    Reply = {error, running},
+    {reply, Reply, running, State}.
+
+%% @private
+%% @doc Not used.
+handle_event(_Event, StateName, State) ->
+    {next_state, StateName, State}.
+
+%% @private
+%% @doc
+handle_sync_event({set_interval, Initerval}, _From, StateName, State) ->
+    NewState0 = maybe_cancel_timer(State#gc_manager_state{interval=Initerval}),
+    NewState = schedule_next(NewState0),
+    {reply, ok, StateName, NewState};
+handle_sync_event(status, _From, StateName, #gc_manager_state{gc_batch_pid=Pid} = State) ->
+    {_GCDStateName, GCDState} = maybe_current_state(Pid),
+    NewState = State#gc_manager_state{current_batch=GCDState},
+    {reply, {ok, {StateName, NewState}}, StateName, NewState};
+handle_sync_event(stop, _, _, State) ->
+    %% for tests
+    {stop, normal, ok, State};
+handle_sync_event(_Event, _From, StateName, State) ->
+    Reply = {error, illegal_sync_send_all_state_call},
+    {reply, Reply, StateName, State}.
+
+%% @private
+%% @doc Because process flag of trap_exit is true, gc_batch
+%% failure will be delivered as a message.
+handle_info({'EXIT', _Pid, normal}, StateName, State) ->
```

Comment:
- No logging in this clause?
- If it is in `running` state, transition to `idle` is needed?